src/test/java/com/fasterxml/jackson/databind/struct/UnwrappedPropertyBasedCreatorWithPrefixTest.java

```diff
@@ -0,0 +1,42 @@
+package com.fasterxml.jackson.databind.struct;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonUnwrapped;
+import com.fasterxml.jackson.databind.BaseMapTest;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class UnwrappedPropertyBasedCreatorWithPrefixTest extends BaseMapTest
+{
+    static class Outer {
+        @JsonUnwrapped(prefix = "inner-")
+        Inner inner;
+    }
+
+    static class Inner {
+        private final String _property;
+
+        public Inner(@JsonProperty("property") String property) {
+            _property = property;
+        }
+
+        public String getProperty() {
+            return _property;
+        }
+    }
```

Comment:
@cowtowncoder While looking at something else, I noticed that you could not do this because the `NameTransformer` used for unwrapping is not applied to `_propertyBasedCreator`, only to `_beanProperties`. This commit seems to fix that: https://github.com/FasterXML/jackson-databind/pull/4271/commits/d65dfacf8d9ec371910db8f230b775e11828b7c1. I guess it broadly fits the scope of the PR, but I can also remove it and make a separate one. In theory its more or less unrelated.